tests/typecheck/managers/querysets/test_custom_queryset.yml

```diff
@@ -0,0 +1,42 @@
+- case: test_custom_queryset_with_passthrough_values_list
+  main: |
+    from typing import Any, TypeVar, Self
+    from django.db.models.base import Model
+    from django.db.models.query import QuerySet
+    from myapp.models import MyUser
+
+    _Model = TypeVar("_Model", bound=Model, covariant=True)
+
+    class CustomQuerySet(QuerySet[_Model]):
+        def values_list(self, *args: Any, **kwargs: Any) -> QuerySet[_Model]:
+            return super().values_list(*args, **kwargs)
```

Comment:
stupid question: what will happen if we do `kwargs.pop('flat'); super().values_list(*args, flat=True, **kwargs)`?

Reply:
The `flat=True` doesn't change the return type of the `get` call in the tests. This seems like a separate bug and is beyond my knowledge of mypy or mypy plugins - seems like a tricky one to solve.